src/isomorphicClerk.ts

```diff
@@ -765,14 +772,18 @@ export class IsomorphicClerk implements IsomorphicLoadedClerk {
   }
 
   addListener = (listener: ListenerCallback): UnsubscribeCallback => {
-    const callback = () => this.clerkjs?.addListener(listener)
-
     if (this.clerkjs) {
-      return callback() as UnsubscribeCallback
-    }
-    else {
-      this.premountMethodCalls.set('addListener', callback)
-      return () => this.premountMethodCalls.delete('addListener')
+      return this.clerkjs.addListener(listener)
+    } else {
+      const unsubscribe = () => {
+        const item = this.premountAddListeners.get(listener)
+        if (item) {
+          if (item.clerkUnsubscribe) item.clerkUnsubscribe()
+          this.premountAddListeners.delete(listener)
+        }
+      }
+      this.premountAddListeners.set(listener, { listener, unsubscribe })
+      return unsubscribe
```

Comment:
Ah, I see what you mean now. Since `addListener` also gets added to premounted method calls, there are timing issues when using different listeners throughout the app, is that correct?

Reply:
Yea , since there can be multiple callled either internally in the plugin or externally via our app. you have to track all of them that were called before the script was loaded and rewire the unsubscribe once the get hooked up post hydration .

Reply:
Thanks for answering!